lib/http/features/auto_inflate.rb

```diff
@@ -3,12 +3,20 @@
 module HTTP
   module Features
     class AutoInflate < Feature
-      def stream_for(connection, response)
-        if %w[deflate gzip x-gzip].include?(response.headers[:content_encoding])
-          Response::Inflater.new(connection)
-        else
-          connection
-        end
+      def wrap_response(response)
+        return response unless %w[deflate gzip x-gzip].include?(response.headers[:content_encoding])
+        Response.new(
+          :status => response.status,
+          :version => response.version,
+          :headers => response.headers,
+          :proxy_headers => response.proxy_headers,
+          :connection => response.connection,
+          :body => stream_for(response.connection)
+        )
+      end
+
+      def stream_for(connection)
+        Response::Body.new(Response::Inflater.new(connection))
```

Comment:
Do you think it would make more sense for the `Inflater` to wrap the `Response::Body` object instead of `Connection`? That would be symmetrical to `CompressedBody` wrapping the `Request::Body` object. And it makes more sense to me that `Response::Body` object reads the response body as is, and then the `Inflater` wrapper changes it.

Reply:
Probably. In fact, I think the entire `Inflater` methods could be moved into this class, similar to how AutoDeflate does it. But, again, I wanted this PR to remain low-impact.

Personally, I'm more interested in being able to add more "features" like these as plugins, like logging/instrumentation or caching. I was reading through the PR history and debating how to monkeypatch it for my own use, when I noticed that with a few minor changes to how the AutoInflate/Deflate features were implemented, then `#use` could become much more powerful, and enable hooks that I need without moneypatching.

I'd like for this to get merged, and then perhaps used as a jumping-off point for further refactorings and more features. There's also some ergonomic low-hanging-fruit here too (like implementing a `Request#merge(opts)` method, rather than each feature implementation having to reconstruct the hash by hand. For example, in this file L8-15 could be replaced with `response.merge(body: stream_for(response.connection))`.

Reply:
Yeah, that makes total sense.  I do like the gradual approach to refactoring, with keeping the same functionality 👍